### ISSUE TITLE
Add ability to crack the world gen seed

### ISSUE THREAD

**Issue by ZenithRogue** [Earthcomputer/clientcommands#85]
Basically client-sided seed cracking

Making this issue to mitigate seed cracking discussion from issue #56

**Comment by Earthcomputer** [comment]
To clarify, this is talking about world gen seed cracking.

**Comment by Earthcomputer** [comment]
If this is added, it will likely be something that runs in the background while you're playing, and won't aim to be anywhere near as fast as Neil's seed cracker.

**Comment by hube12** [comment]
@Earthcomputer if this is added it will not bruteforce but use my formulas. Ok i am giving myself a deadline: by Xmas 2k19 this will be implemented somehow here

**Comment by MCRcortex** [comment]
@hube12 by your formulas you mean using the end to crack it?

**Comment by hube12** [comment]
@MCRcortex oh no i am talking about lattices but in a light way to accomodate for variation so its upgradable to any version easily

**Comment by MCRcortex** [comment]
Oooooo  i really wanna see this

**Comment by ZenithRogue** [comment]
Sounds good to me.

**Comment by Anunayj** [comment]
This will be epic if this works.

**Comment by Mhowser** [comment]
If Dream from YouTube can do it solely based off video footage, I have no doubts that we can do this.

**Comment by MCRcortex** [comment]
dude, dream didnt do it. it was monkeys

**Comment by Mhowser** [comment]
True, his team did it. What I saying is that it was done mainly using video footage. However, We would likely have direct access to a server where we can precisely measure key elements needed to crack a world seed.

**Comment by hube12** [comment]
@Mhowser lol, you dont know what you are talking about xD, WE are the guys that did it ;p

**Comment by Mhowser** [comment]
Well, it's been a while since I saw the video, I'm sorry that I didn't
remember the people who did it.

**Comment by KaptainWutax** [comment]
> True, his team did it.

I didn't know I was part of this guy's team. 🤔

**Comment by Anunayj** [comment]
Its a pity how people Just associate the work of many individuals as complete work of 1 individual who posted a video, even if he contributed not a shit to the project.

**Comment by Mhowser** [comment]
Any progress on this?

**Comment by hube12** [comment]
@Mhowser as promised the seed cracking was implemented as a separate mod here : https://github.com/KaptainWutax/SeedCracker
There is an alpha release that work for 1.15, you can also build one for 1.13 and 1.14 by just going back in the commits

**Comment by Earthcomputer** [comment]
On my end, perhaps all I have to do is add a script interface.
Using [multiconnect](https://github.com/Earthcomputer/multiconnect), you can connect to 1.13 and 1.14 servers from (soon) a 1.15 client (currently 1.14.4). The multiconnect API gives you a way of querying the server version, perhaps there would be a way of integrating this into the seed cracker?

**Comment by hube12** [comment]
@Earthcomputer sadly there was some modification to biomes (they now hash the seed and use it to apply to the last layer (voronoi) before sending to the client) so using multiconnect directly is not possible but with a few tweaks, for sure. Anyway since the mod is a one time use i think its best as it is.

**Comment by leumasme** [comment]
damn, this is epic. prebuilt versions for 1.13/1.14 on the github would be nice tho @hube12

**Comment by ZenithRogue** [comment]
How does the actual crack work @hube12?

**Comment by Earthcomputer** [comment]
I haven't watched the video yet, but I'm assuming he answers that in the video he just made:
https://youtu.be/1ChmLi9og8Q

**Comment by ZenithRogue** [comment]
Just watched it. You load in biomes and structures, and head over into the End. It uses the pillars to finalize the search

**Comment by hube12** [comment]
yep ;)

**Comment by ZenithRogue** [comment]
Anyway you can do it without the end?

**Comment by xpple** [comment]
Out of scope for clientcommands to implement. There are other mods that can do this.
